include/pagmo/algorithm.hpp

```diff
@@ -640,6 +668,58 @@ class PAGMO_DLL_PUBLIC algorithm
     // Check if the algorithm is valid.
     bool is_valid() const;
 
+    // Get the type at runtime.
+    std::type_index get_type_index() const;
+
+    /// Get a const pointer to the UDA.
+    /**
+     * \verbatim embed:rst:leading-asterisk
+     * .. versionadded:: 2.15
+     *
+     * This function will return a raw const pointer
+     * to the internal UDA instance. Differently from
+     * :cpp:func:`~pagmo::algorithm::extract()`, this function
+     * does not require to pass the correct type
+     * in input. It is however the user's responsibility
+     * to cast the returned void pointer to the correct type.
+     *
+     * .. note::
+     *
+     *    The returned value is a raw non-owning pointer: the lifetime of the pointee is tied to the lifetime
+     *    of ``this``, and ``delete`` must never be called on the pointer.
+     * \endverbatim
+     *
+     * @return a pointer to the internal UDA.
+     */
+    const void *get_ptr() const;
+
+    /// Get a mutable pointer to the UDA.
+    /**
+     * \verbatim embed:rst:leading-asterisk
+     * .. versionadded:: 2.15
+     *
+     * This function will return a raw pointer
+     * to the internal UDA instance. Differently from
+     * :cpp:func:`~pagmo::algorithm::extract()`, this function
+     * does not require to pass the correct type
+     * in input. It is however the user's responsibility
+     * to cast the returned void pointer to the correct type.
+     *
+     * .. note::
+     *
+     *    The returned value is a raw non-owning pointer: the lifetime of the pointee is tied to the lifetime
+     *    of ``this``, and ``delete`` must never be called on the pointer.
```

Comment:
What happens if delete IS called? Can you just add a brief explanation?

Reply:
Undefined behaviour of course :) This is just the standard warning that is there on all ``extract()`` functions.